### DISCUSSION TITLE
Can PiVPN use DNS over HTTPS/DNS over TLS?

### DISCUSSION THREAD

**Discussion by Epikest** [pivpn/pivpn#1466] [Support] [answered]
If I had a DoH/DoT client running on the server, would it be possible to use the forwarded DNS queries from the DoH/DoT client instead of the system's DNS?

**Comment by orazioedoardo** [comment, marked as answer]
Isn't this the same as https://github.com/pivpn/pivpn/discussions/1463?

**Reply by Epikest** [reply]
That one is for system DNS.

**Reply by orazioedoardo** [reply]
So instead of using nextdns indirectly on the pivpn client by running nextdns on the pivpn server you want to use nextdns on the client directly without going through the pi?

**Reply by Epikest** [reply]
I want NextDNS to run on the Pi but with a DoH/DoT client like https://github.com/LinkTed/doh-client.

**Reply by orazioedoardo** [reply]
You just need to have that DNS client listen on 0.0.0.0:53 (like nextdns on the previous discussion) or at minimum LAN IP:53 or VPN IP:53 and point the vpn client to it using LAN IP or VPN IP, yours was 10.6.0.1. You can replace that nextdns daemon with any dns server you want: dnsmasq, unbound, pihole, dns over pigeon, whatever.
